database/migrations/2024_06_18_174043_data_seed_remove_analytics_resource_role.php

```diff
@@ -34,37 +34,40 @@
 </COPYRIGHT>
 */
 
-namespace AdvisingApp\Analytics\Filament\Resources\AnalyticsResourceCategoryResource\Pages;
+use Illuminate\Support\Str;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Migrations\Migration;
 
-use Filament\Actions\EditAction;
-use Filament\Infolists\Infolist;
-use Filament\Resources\Pages\ViewRecord;
-use Filament\Infolists\Components\Section;
-use Filament\Infolists\Components\TextEntry;
-use AdvisingApp\Analytics\Filament\Resources\AnalyticsResourceCategoryResource;
-
-class ViewAnalyticsResourceCategory extends ViewRecord
-{
-    protected static string $resource = AnalyticsResourceCategoryResource::class;
-
-    public function infolist(Infolist $infolist): Infolist
+return new class () extends Migration {
+    public function up(): void
     {
-        return $infolist
-            ->schema([
-                Section::make()
-                    ->schema([
-                        TextEntry::make('name'),
-                        TextEntry::make('classification'),
-                        TextEntry::make('description'),
-                    ])
-                    ->columns(),
-            ]);
+        $role_details = DB::table('roles')->where('name', 'analytics.analytics_management')->whereIn('guard_name', ['web', 'api'])->get();
+
+        if (! $role_details->isEmpty()) {
+            foreach ($role_details as $role) {
+                DB::table('model_has_roles')->where('role_id', $role->id)->delete();
+                DB::table('role_has_permissions')->where('role_id', $role->id)->delete();
+                DB::table('roles')->where('name', $role->name)->delete();
```

Comment:
```suggestion
                DB::table('roles')->where('id', $role->id)->delete();
```
Please make sure to test this change.

Without doing so the first pass of this will delete both roles

Reply:
Hi @Orrison 
Changes done and I also tested things its working fine for me. 
see this screen shot there is two roles after run rollback migration.
![image](https://github.com/canyongbs/advisingapp/assets/170341685/10ce3c69-ad35-45ff-a986-797b7c916927)
See after again run migration, two roles deleted.
![image](https://github.com/canyongbs/advisingapp/assets/170341685/126c179f-28f3-47c1-a1f0-f54424c811d3)